rust/nix-util/src/string_return.rs

```diff
@@ -16,6 +16,10 @@ pub unsafe extern "C" fn callback_get_vec_u8(
     (*ret).extend_from_slice(slice);
 }
 
+pub fn callback_get_vec_u8_data(vec: &mut Vec<u8>) -> *mut std::os::raw::c_void {
+    vec as *mut Vec<u8> as *mut std::os::raw::c_void
+}
+
```

Comment:
Nice. This makes the interface much cleaner.

Now that I see it like this though, I had a thought: can we directly pass in a `&mut String` casted to `*mut c_void` as the user data, and then cast it back in the function and do all the post-processing in there so we don't need this every time?
```
    String::from_utf8(raw_buffer).map_err(|e| e.into())
```

Maybe we can't because we have no way of dealing with the possible error...